### PR TITLE
zuul-core: use non deprecated Http2 Multiplex handlers

### PR DESCRIPTION
**Pull request by carl-mastrangelo**

### CODE REVIEW

**Comment by carl-mastrangelo** [comment]
cc: @philrud

File: zuul-core/src/main/java/com/netflix/zuul/netty/server/http2/Http2OrHttpHandler.java [hunk 1]
```diff
@@ -44,7 +43,6 @@
  */
 public class Http2OrHttpHandler extends ApplicationProtocolNegotiationHandler {
     public static final AttributeKey<String> PROTOCOL_NAME = AttributeKey.valueOf("protocol_name");
-    public static final AttributeKey<Http2Connection> H2_CONN_KEY = AttributeKey.newInstance("h2_connection");
```
Comment (artgon) [inline zuul-core/src/main/java/com/netflix/zuul/netty/server/http2/Http2OrHttpHandler.java]: What happened to the `H2_CONN_KEY` attribute?

Comment (carl-mastrangelo) [reply zuul-core/src/main/java/com/netflix/zuul/netty/server/http2/Http2OrHttpHandler.java]: It was never used internally or externally, so I removed it.   The connection is not accessible in the new API as well.

**Comment by carl-mastrangelo** [comment]
The origin side will follow up shortly.

**Pull request merged**